static/js/place/chart.tsx

```diff
@@ -652,33 +649,54 @@ class Chart extends React.Component<ChartPropType, ChartStateType> {
         }
         break;
       case chartTypeEnum.RANKING:
-        if (this.props.rankingChartData) {
-          this.props.rankingChartData
-            .then((rankingChartData) => {
-              if (_.isEmpty(this.props.statsVars)) {
-                this.setState({ display: false });
-                return;
-              }
-              const sv = this.props.statsVars[0];
-              const svData = rankingChartData[sv];
-              // Do not display the ranking chart if total data points is less than the MIN_RANKING_DATAPOINTS
-              if (
-                _.isEmpty(svData) ||
-                svData.numDataPoints < MIN_RANKING_DATAPOINTS
-              ) {
-                this.setState({ display: false });
-                return;
-              }
-              for (const data of svData.data) {
-                data.value = data.value * scaling;
-              }
-              svData.rankingData = this.getRankingChartData(svData);
-              this.setState({ rankingChartDataGroup: svData });
-            })
-            .catch(() => {
+        // Fields like unit, denom, scaling and etc. are not used for data
+        // fetch, hencing setting a dummy value here.
+        fetchData({
+          id: "",
+          enclosedPlaceType: this.props.rankingPlaceType,
+          place: { dcid: this.props.parentPlaceDcid, name: "", types: [] },
+          rankingMetadata: {
+            diffBaseDate: "",
+            showHighest: true,
+            showLowest: true,
+            showMultiColumn: false,
+          },
+          statVarSpec: this.props.statsVars.map((x) => {
+            return {
+              denom: "",
+              log: false,
+              scaling: 1,
+              statVar: x,
+              unit: "",
+            };
+          }),
+          title: "",
+        })
+          .then((rankingChartData) => {
+            if (_.isEmpty(this.props.statsVars)) {
```

Comment:
we don't even need to fetch data if this.props.statsVars is empty right? so could move this before the fetchData

Reply:
Done